### PR TITLE
hash-cache-tool: Scan files with mmap and bins

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

The accounts hash cache tool can be faster by scanning mmap'd files, and then optionally binning the results by pubkey. These binned results will be used when constructing the full state to diff in the next PR.

#### Summary of Changes

Scanning entries uses mmap'd files into bins.

### CODE REVIEW

File: accounts-db/src/pubkey_bins.rs [hunk 1]
```diff
@@ -11,12 +11,12 @@ impl PubkeyBinCalculator24 {
         std::mem::size_of::<T>() * 8
     }
 
-    pub(crate) fn log_2(x: u32) -> u32 {
+    pub fn log_2(x: u32) -> u32 {
```
Comment (jeffwashington) [inline accounts-db/src/pubkey_bins.rs]: doesn't probably matter, but is this necessary?

**Review by jeffwashington — DISMISSED** [review]
lgtm

File: accounts-db/src/pubkey_bins.rs (hunk 1 above)
Comment (brooksprumo) [reply accounts-db/src/pubkey_bins.rs]: No, not necessary. Let me revert that.

Comment (brooksprumo) [reply accounts-db/src/pubkey_bins.rs]: Fixed in https://github.com/anza-xyz/agave/pull/2504/commits/f0a2790f963d58ad05407faa4e383f82440421a5.

**Review by jeffwashington — APPROVED** [review]
lgtm

**Review by HaoranYi — COMMENTED** [review]
lgtm.

**Pull request merged**